templates/workloads/partials/cf/subscribe.yml

```
@@ -0,0 +1,176 @@
+{{- if .Subscribe}}
+EventsQueue:
+  Type: AWS::SQS::Queue
+  {{- if .Subscribe.Queue}}
+  Properties:
+    KmsMasterKeyId: 'alias/aws/sqs'
+    {{- if .Subscribe.Queue.Retention}}
+    MessageRetentionPeriod: {{.Subscribe.Queue.Retention}}
+    {{- end}}
+    {{- if .Subscribe.Queue.Delay}}
+    DelaySeconds: {{.Subscribe.Queue.Delay}}
+    {{- end}}
+    {{- if .Subscribe.Queue.Timeout}}
+    VisibilityTimeout: {{.Subscribe.Queue.Timeout}}
+    {{- end}}
+    {{- if .Subscribe.Queue.FIFO}}
+    FifoQueue: true
+    {{- if .Subscribe.Queue.FIFO.HighThroughput}}
+    FifoThroughputLimit: 'perMessageGroupId'
+    DeduplicationScope: 'messageGroup'
+    {{- end}}
+    {{- end}}
+    {{- if .Subscribe.Queue.DeadLetter}}
+    RedrivePolicy:
+      deadLetterTargetArn: !GetAtt DeadLetterQueue.Arn
+      maxReceiveCount: {{.Subscribe.Queue.DeadLetter.Tries}}
+    {{- end}}
+  {{- end}}
+
+{{- if .Subscribe.Queue}}{{- if .Subscribe.Queue.DeadLetter}}
+DeadLetterQueue:
+  Type: AWS::SQS::Queue
+
+DeadLetterPolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref 'DeadLetterQueue']
+    PolicyDocument:
+      Version: "2012-10-17"
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt EventsQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+{{- end}}{{- end}}
+
+QueuePolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref 'EventsQueue']
+    PolicyDocument:
+      Version: '2012-10-17'
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt EventsQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+        {{- range $topic := .Subscribe.Topics}}
+        {{- if not $topic.Queue}}
+        - Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:SendMessage
+          Resource: !GetAtt EventsQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Join ['', [!Sub 'arn:${AWS::Partition}:sns:${AWS::Region}:${AWS::AccountId}:', !Ref AppName, '-', !Ref EnvName, '-{{logicalIDSafe $topic.Service}}-{{logicalIDSafe $topic.Name}}']]
```

I think we want to preserve the service name without replacing the dashes:
```suggestion
              aws:SourceArn: !Join ['', [!Sub 'arn:${AWS::Partition}:sns:${AWS::Region}:${AWS::AccountId}:', !Ref AppName, '-', !Ref EnvName, '-{{$topic.Service}}-{{logicalIDSafe $topic.Name}}']]
```
while refering to the arn since the stackname for services follow the format: `{{.App}}-{{.Env}}-{{.Service}}` with no modifications